openfe/protocols/openmm_septop/base.py

```diff
@@ -0,0 +1,1300 @@
+        reporter : multistate.MultiStateReporter
+          The reporter associated with the sampler.
+        settings : dict[str, SettingsBaseModel]
+          The dictionary of settings for the protocol.
+        dry : bool
+          Whether or not to dry run the simulation
+
+        Returns
+        -------
+        unit_results_dict : Optional[dict]
+          A dictionary containing all the free energy results,
+          if not a dry run.
+        """
+        # Get the relevant simulation steps
+        mc_steps = settings_validation.convert_steps_per_iteration(
+            simulation_settings=settings['simulation_settings'],
+            integrator_settings=settings['integrator_settings'],
+        )
+
+        equil_steps = settings_validation.get_simsteps(
+            sim_length=settings['simulation_settings'].equilibration_length,
+            timestep=settings['integrator_settings'].timestep,
+            mc_steps=mc_steps,
+        )
+        prod_steps = settings_validation.get_simsteps(
+            sim_length=settings['simulation_settings'].production_length,
+            timestep=settings['integrator_settings'].timestep,
+            mc_steps=mc_steps,
+        )
+
+        if not dry:  # pragma: no-cover
+            # minimize
+            if self.verbose:
+                self.logger.info("minimizing systems")
+            sampler.minimize(
+                max_iterations=settings[
+                    'simulation_settings'].minimization_steps
+            )
+            # equilibrate
+            if self.verbose:
+                self.logger.info("equilibrating systems")
+
+            sampler.equilibrate(int(equil_steps / mc_steps))  # type: ignore
+
+            # production
+            if self.verbose:
+                self.logger.info("running production phase")
+            sampler.extend(int(prod_steps / mc_steps))  # type: ignore
+
+            if self.verbose:
+                self.logger.info("production phase complete")
+
+            if self.verbose:
+                self.logger.info("post-simulation result analysis")
+
+            analyzer = multistate_analysis.MultistateEquilFEAnalysis(
+                reporter,
+                sampling_method=settings[
+                    'simulation_settings'].sampler_method.lower(),
+                result_units=unit.kilocalorie_per_mole
+            )
+            analyzer.plot(filepath=self.shared_basepath, filename_prefix="")
+            analyzer.close()
+
+            return analyzer.unit_results_dict
+
+        else:
+            # close reporter when you're done, prevent file handle clashes
+            reporter.close()
+
+            # clean up the reporter file
+            fns = [self.shared_basepath / settings[
+                'output_settings'].output_filename,
+                   self.shared_basepath / settings[
+                       'output_settings'].checkpoint_storage_filename]
+            for fn in fns:
+                os.remove(fn)
```

Comment:
Not critical but it looks like os is imported just for this, we could just do this with pathlib and remove the import.